### PR TITLE
NET-4992- Kitchen sink gateway test 

### PR DESCRIPTION
**Pull request by sarahalsmiller**
Changes proposed in this PR:
- Added acceptance test to utilize several API GW features at once. Potentially as time goes on we will need to add additional asserts, feature flags to this test to keep it up to date 

How I've tested this PR:
- Run the acceptance test locally

How I expect reviewers to test this PR:
- CI test pass

Checklist:
- [ X ] Tests added
- [ ] [CHANGELOG entry added](https://github.com/hashicorp/consul-k8s/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

### CODE REVIEW

File: acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+	"github.com/hashicorp/consul-k8s/acceptance/framework/helpers"
+	"github.com/hashicorp/consul-k8s/acceptance/framework/logger"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/sdk/testutil/retry"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/types"
+	gwv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// Enabled everything possible, see if anything breaks.
```
Comment (missylbytes) [inline acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go:27]: 🔥

File: acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+
+	gatewayParametersRef := &gwv1beta1.ParametersReference{
+		Group: gwv1beta1.Group(v1alpha1.ConsulHashicorpGroup),
+		Kind:  gwv1beta1.Kind(v1alpha1.GatewayClassConfigKind),
+		Name:  gatewayClassConfigName,
+	}
+
+	// Create gateway class referencing gateway-class-config.
+	logger.Log(t, "creating controlled gateway class")
+	createGatewayClass(t, k8sClient, gatewayClassName, gatewayClassControllerName, gatewayParametersRef)
```
Comment (jm96441n) [inline acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go:123]: do we still need to create the gatewayclass and all that manually?

**Review by jm96441n — APPROVED** [review]
minor comment but overall LGTM

File: acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go (hunk 2 above)
Comment (missylbytes) [reply acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go:123]: I think `managedGatewayClass` defaults to true so we should be able to skip these steps if there isn't anything special about the `classConfig`.

Comment (sarahalsmiller) [reply acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go:123]: I could swap it over to kustomize before I merge it in for sure. I can't remember why it was split out this way.

Comment (sarahalsmiller) [reply acceptance/tests/api-gateway/api_gateway_kitchen_sink_test.go:123]: `kitchen-sink-gateway-class-config` does have configuration that utilities the min/max deployment config.

**Pull request merged**